### PR TITLE
Initial iteration of the React Map component

### PR DESCRIPTION
**Pull request by mendozaline**
This is the basic map component for issue [#62](https://github.com/hackoregon/civic/issues/62). I added react-map-gl, created a new React Map component and story for it. 

I want to get this reviewed before moving on to the deck.gl overlays.

A testing file still needs to be added and written. I'll need help with that.

Some questions that we still need to resolve include what to do about a default map style and any other defaults we want to set.

### CODE REVIEW

File: packages/component-library/src/ReactMap/ReactMap.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import React, { PropTypes } from 'react';
+import MapGL from 'react-map-gl';
+import './mapbox-gl.css';
+
+const ReactMap = (props) => {
```
Comment (cp) [inline packages/component-library/src/ReactMap/ReactMap.js]: Might make sense to simplify the name of this component (and file) to just `Map`. Thoughts?

Comment (mendozaline) [reply packages/component-library/src/ReactMap/ReactMap.js]: That makes sense. I was trying to differentiate it from the Leaflet Map component already there. I'll rename it.

**Comment by mendozaline** [comment]
I renamed the component from ReactMap to BaseMap and converted it from a functional to class component. This allows the map to now pan, zoom, and resize.

I also included and wrote tests for the component.

File: packages/component-library/src/BaseMap/BaseMap.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  margin: auto;
+  max-width: 900px;
+`;
+
+export default class BaseMap extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      viewport:{
+        width: window.innerWidth > 900 ? 898 : window.innerWidth,
```
Comment (jaronheard) [inline packages/component-library/src/BaseMap/BaseMap.js]: May change this to work with responsive containers if it doesn't already

Comment (mendozaline) [reply packages/component-library/src/BaseMap/BaseMap.js]: I was meaning to change that. It's already inside a div and it's responsive, so the map resize when the window changes and stays centered. But I'll update it.

**Pull request merged**